lib/index.js

```diff
@@ -153,36 +153,72 @@ export function configureUnifiedLanguageServer(
    * @returns {Promise<VFile[]>}
    */
   function processDocuments(textDocuments, alwaysStringify = false) {
-    return new Promise((resolve, reject) => {
-      engine(
-        {
-          alwaysStringify,
-          files: textDocuments.map((document) => lspDocumentToVfile(document)),
-          ignoreName,
-          packageField,
-          pluginPrefix,
-          plugins,
-          processor,
-          quiet: false,
-          rcName,
-          silentlyIgnore: true,
-          streamError: new PassThrough(),
-          streamOut: new PassThrough()
-        },
-        (error, _, context) => {
-          // An error never occur and can’t be reproduced. Thus us ab internal
-          // error in unified-engine. If a plugin throws, it’s reported as a
-          // vfile message.
-          /* c8 ignore start */
-          if (error) {
-            reject(error)
-          } else {
-            resolve((context && context.files) || [])
-          }
-          /* c8 ignore stop */
-        }
-      )
-    })
+    // LSP uses `file:` URLs (hrefs), `unified-engine` expects a paths.
+    // `process.cwd()` does not add a final slash, but `file:` URLs often do.
+    const workspacesAsPaths = workspaces
+      ? workspaces.map((d) => fileURLToPath(d).replace(/\/$/, ''))
+      : [process.cwd()]
```

Comment:
I’m assuming that actual LSP clients will send workspaces, or at least `rootUri`.
The tests don’t to keep things simple, and I think the default of `process.cwd()` would be fine.